### PR TITLE
Top contributors SME + AIE (data parts)

### PR DESCRIPTION
**Pull request by i-kushlianski**
- Top Contributors SME data https://www.pivotaltracker.com/story/show/169728620
- Top Contributors AIE data https://www.pivotaltracker.com/story/show/169890873 (WIP)

### CODE REVIEW

File: packages/bootstrap/src/components/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -65,7 +67,7 @@ describe('Dashboard Widgets', () => {
             provider={provider}
             config={widgetConfig}
           >
-            <TopContributors />
+            <TopTeachingContributors />
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/__tests__/index.spec.tsx]: In my opinion, we have to test here `ContributorsPresenter`

Comment (alex-melnik) [reply packages/bootstrap/src/components/__tests__/index.spec.tsx]: Without `act`

File: packages/bootstrap/src/components/dashboard/hooks/useJustBeforeTodayMidnight.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const MS_IN_DAY = 86400000;
+
+export const useJustBeforeTodayMidnight = () => {
+  const dateNow = Date.now();
+
+  return dateNow - (dateNow % MS_IN_DAY) - 100;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/hooks/useJustBeforeTodayMidnight.ts]: Cool! Would be nice if the hook returns also yesterday value

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/presenters/ContributorAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { ContributorProps } from './Presenter';
+
+export const ContributorAvatar: React.FC<ContributorProps> = ({
+  userId,
+  ranking,
+  graphUrl,
+}) => (
+  <AvatarWrapper>
+    <ContributorRanking>{ranking}</ContributorRanking>
+    <AvatarImage src={`${graphUrl}/${userId}/picture`} />
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/presenters/ContributorAvatar.tsx]: Why not to configure url in Data Layer?

File: packages/bootstrap/src/components/dashboard/hooks/useJustBeforeTodayMidnight.ts (hunk 2 above)
Comment (i-kushlianski) [reply packages/bootstrap/src/components/dashboard/hooks/useJustBeforeTodayMidnight.ts]: I can return { todayMidnight, yesterdayMidnight }

File: packages/bootstrap/src/components/__tests__/index.spec.tsx (hunk 1 above)
Comment (i-kushlianski) [reply packages/bootstrap/src/components/__tests__/index.spec.tsx]: done

File: packages/bootstrap/src/components/dashboard/widgets/Welcome/Welcome.tsx [hunk 4]
```diff
@@ -98,7 +98,7 @@ export const Welcome: React.FC = () => {
   return (
     <Box centered>
       {content}
-      <Loader active={loading} />
+      {/*<Loader active={loading} />*/}
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Welcome/Welcome.tsx]: Do we need to remove it?

File: packages/bootstrap/src/components/dashboard/widgets/TopContributors/presenters/Presenter.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  );
+};
+
+const getWidgetContent = ({
+  contributorsColumns,
+  error,
+  loading,
+}: GetWidgetContent) => {
+  if (error) {
+    return <Message>Something went wrong</Message>;
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/TopContributors/presenters/Presenter.tsx]: Can we also log the error for dev purposes?

File: packages/bootstrap/src/components/dashboard/types.ts [hunk 6]
```diff
@@ -116,3 +118,16 @@ export interface DetailsContainerItem {
...
+
+/**
+ * Top Contributors Widget
+ */
+export interface ContributorProps {
+  userId?: string;
+  ranking?: number;
+  count?: number;
+  avatarUrl?: string;
+}
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/types.ts]: Why optional?

File: packages/bootstrap/src/components/dashboard/widgets/Welcome/Welcome.tsx (hunk 4 above)
Comment (i-kushlianski) [reply packages/bootstrap/src/components/dashboard/widgets/Welcome/Welcome.tsx]: No, we certainly don't. I'll bring it back

File: packages/bootstrap/src/components/dashboard/types.ts (hunk 6 above)
Comment (miraage) [reply packages/bootstrap/src/components/dashboard/types.ts]: Aren't those props being always populated?

Comment (i-kushlianski) [reply packages/bootstrap/src/components/dashboard/types.ts]: I will fix it and make them all required

**Pull request merged**